flutter/lib/src/integrations/web_session_integration.dart

```diff
@@ -0,0 +1,97 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+
+import '../../sentry_flutter.dart';
+import '../native/sentry_native_binding.dart';
+import '../web/web_session_handler.dart';
+
+/// Integration for handling web sessions in Sentry.
+///
+/// Enables tracking of web sessions in a manner similar to page views.
+/// It requires using the[SentryNavigatorObserver] since sessions are automatically
+/// started on route changes and updated when errors occur.
+///
+/// The integration is only active on web platforms with enableAutoSessionTracking enabled.
+class WebSessionIntegration implements Integration<SentryFlutterOptions> {
+  static const integrationName = 'WebSessionIntegration';
+  final SentryNativeBinding _native;
+  SentryFlutterOptions? _options;
+  BeforeSendEventObserver? _observer;
+  WebSessionHandler? _webSessionHandler;
+  WebSessionHandler? get webSessionHandler => _webSessionHandler;
+  bool _isEnabled = false;
+
+  WebSessionIntegration(this._native);
+
+  @override
+  void call(Hub hub, SentryFlutterOptions options) {
+    _options = options;
+
+    _options?.logger(SentryLevel.info,
+        '$integrationName initialization started, waiting for SentryNavigatorObserver to be initialized.');
+  }
+
+  @override
+  void close() {
+    if (_options != null && _observer != null) {
+      _options!.removeBeforeSendEventObserver(_observer!);
+    }
+  }
+
+  /// [SentryNavigatorObserver] is created at a later point than the integration
+  /// so we need to wait until this function is called by the observer.
+  void enable() {
+    if (_isEnabled) {
+      _options?.logger(
+          SentryLevel.debug, '$integrationName is already enabled.');
+      return;
+    }
+    if (!_shouldEnable()) {
+      return;
+    }
+
+    _webSessionHandler = WebSessionHandler(_native);
+    _observer = WebSessionUpdateObserver(_webSessionHandler!);
+    _options?.addBeforeSendEventObserver(_observer!);
+    _options?.sdk.addIntegration(integrationName);
+    _isEnabled = true;
```

Comment:
We could make `_isEnabled` a computed property, since it is only set when an `_webSessionHandler` is present.